pkg/controller/controller.go

```diff
@@ -188,6 +197,20 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 	c.instanceQueue.ShutDown()
 	c.bindingQueue.ShutDown()
 	c.pollingQueue.ShutDown()
+
+	for i := 0; i < totalNumberOfWorkers; i++ {
+		<-done
+	}
+}
+
+// createWorker creates and runs a worker thread that just processes items in the
+// specified queue. The worker will run until stopCh is closed. When the worker
+// stops, a token will be added to the done channel.
+func createWorker(queue workqueue.RateLimitingInterface, resourceType string, maxRetries int, forgetAfterSuccess bool, reconciler func(key string) error, stopCh <-chan struct{}, done chan<- struct{}) {
```

Comment:
Why an internal goroutine vs calling `go createWorker` ?

Reply:
No reason other than that it was more code at the caller side. Is one preferred over the other?

Reply:
I think the fix is correct, but should be implemented with [`WaitGroup`](https://golang.org/pkg/sync/#WaitGroup).

Ideally, there would be a variant of `wait.Until` that took a WaitGroup to call `Add` and `Done` on.  Then in `Run` you could just call `Wait`